src-plugins/medSegmentation/msegAlgorithmPaintToolbox.cpp

```diff
@@ -978,11 +981,7 @@ void AlgorithmPaintToolbox::updateWandRegion(medAbstractImageView * view, QVecto
     }
 
     if(!view->contains(m_maskAnnotationData))
-    {
         view->addLayer(m_maskAnnotationData);
-        setOutputMetadata(m_imageData, m_maskData);
-        medDataManager::instance()->importData(m_maskData);
-    }
```

Comment:
It's actually not a random data since it's your mask;
Problem is, if you want to create two masks from the same volume, with your solution, you save the first mask then you have to rename it otherwise it will be overwritten by the second mask; which is not user-friendly.